fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersReadServiceImpl.java

```diff
@@ -37,15 +37,27 @@ public class ExternalAssetOwnersReadServiceImpl implements ExternalAssetOwnersRe
     private final ExternalAssetOwnersTransferMapper mapper;
 
     @Override
-    public List<ExternalTransferData> retrieveTransferData(Long loanId, String externalLoanId, String transferExternalId) {
-        List<ExternalAssetOwnerTransfer> result = new ArrayList<>();
+    public Page<ExternalTransferData> retrieveTransferData(Long loanId, String externalLoanId, String transferExternalId, Integer offset,
+            Integer limit) {
+        org.springframework.data.domain.Page<ExternalAssetOwnerTransfer> result;
+        if (offset == null) {
+            offset = 0;
+        }
+        if (limit == null) {
+            limit = 100;
+        }
+        PageRequest pageRequest = PageRequest.of(offset, limit);
         if (loanId != null) {
-            result.addAll(externalAssetOwnerTransferRepository.findAllByLoanId(loanId));
+            result = externalAssetOwnerTransferRepository.findAllByLoanId(loanId, pageRequest);
```

Comment:
I think for all these repository calls where we use pageable we are missing a crucial thing, a consistent ordering of the result. Let's do an ID based ordering at least for the query to produce consistent results between runs.